hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterServiceImpl.java

```diff
@@ -233,7 +233,7 @@ void doRemoveAddress(Address deadAddress, boolean destroyConnection) {
             }
             Connection conn = node.connectionManager.getConnection(deadAddress);
             if (destroyConnection && conn != null) {
-                node.connectionManager.destroyConnection(conn);
+                conn.close(reason, null);
```

Comment:
Yes, was just hard to read on GitHub (file order) and of course not comprehensible in IDEA yet (outdated code).